prov/sm2/src/sm2_coordination.c

```diff
@@ -495,6 +522,7 @@ ssize_t sm2_coordinator_unlock(struct sm2_mmap *map) {
 */
 void* sm2_coordinator_extend_for_entry(struct sm2_mmap *map, int last_valid_entry) {
 	size_t new_size;
+	assert(last_valid_entry <= SM2_MAX_PEERS);
```

Comment:
This isn't ok... we can have multiple runs happening at a time (we could make it less than universe_size if that is helpful)